agent/consul/state/catalog_test.go

```diff
@@ -1674,31 +1663,33 @@ func TestStateStore_ServicesByNodeMeta(t *testing.T) {
 		if err != nil {
 			t.Fatalf("err: %s", err)
 		}
-		expected := structs.Services{
-			"redis": []string{"primary", "prod"},
-		}
-		sort.Strings(res["redis"])
-		require.Equal(t, expected, res)
+		require.Len(t, res, 1)
+		require.Condition(t, func() bool {
+			return res[0].ServiceID == "service1" && res[0].Node == "node0"
+		})
 	})
 
 	t.Run("Get all services using the common meta value", func(t *testing.T) {
 		_, res, err := s.ServicesByNodeMeta(ws, map[string]string{"common": "1"}, nil)
 		if err != nil {
 			t.Fatalf("err: %s", err)
 		}
-		expected := structs.Services{
-			"redis": []string{"primary", "prod", "replica"},
-		}
-		sort.Strings(res["redis"])
-		require.Equal(t, expected, res)
+		require.Len(t, res, 2)
+		require.Condition(t, func() bool {
+			// Services from both nodes should be in the resultset
+			return (res[0].Node == "node0" && res[1].Node == "node1") ||
+				(res[0].Node == "node1" && res[1].Node == "node0")
+		})
 	})
 
 	t.Run("Get an empty list for an invalid meta value", func(t *testing.T) {
 		_, res, err := s.ServicesByNodeMeta(ws, map[string]string{"invalid": "nope"}, nil)
 		if err != nil {
 			t.Fatalf("err: %s", err)
 		}
-		expected := structs.Services{}
+		// FIXME: Our changes now return nil instead of empty list, do we care about this change?
+		// (Only the test here caught the difference)
+		var expected []*structs.ServiceNode
```

Comment:
Just highlighting the FIXME here.

Reply:
Good question! This should be ok. A `nil` slice is pretty common and can be passed to `append` , `len`, or `range` without trouble.  It's only a concern for when we serialize a response, and in this case we know we are going to process the slice first, so the RPC response will use a different value.